docs/src/design/many_differentials.md

```diff
@@ -1,113 +1,175 @@
+ChainRules has a system where one primal type (the type having its derivative taken) can have multiple possible differential types (the type of the derivative); and where one differential type can correspond to multiple primal types.
+This is in-contrast to the Swift AD efforts, which has one differential type per primal type. (Swift uses the term associated tangent type, rather than differential type).
 
-This is in-contrast to the Swift AD efforts.
-Which has one differential type, per primal type.
-(Swift uses the term associated tangent type, rather than differential type).
+!!! terminology "differential and associated tangent type"
+    The use of “associated tangent type” in AD is not technically correct, as differentials naturally live in the [_cotangent_ plane](https://en.wikipedia.org/wiki/Cotangent_space) instead of the [tangent plane](https://en.wikipedia.org/wiki/Tangent_space). 
+    However it is often reasonable for AD to treat the cotangent plane and tangent plane as the same thing, and this was an intentional choice by the Swift team.
+    Here we will be a bit more tight with terminology and just stick to the ChainRules terminology and only say “differential type” instead of “tangent type”.
 
-One thing to understand about differentials is they have to form a [vector space](https://en.wikipedia.org/wiki/Vector_space) (or something very like them).
-They need to support addition to each other, they need a zero which doesn't change what it is added to, and they need to support scalar multiplication (this isn't really required, but its handy (e.g. for SGD).
-Beyond being a vector space, differentials need to be able to be added to a primal value to get back another primal value.
+One thing to understand about differentials is they have to form a [vector space](https://en.wikipedia.org/wiki/Vector_space)  (or something very like them).
+They need to support addition to each other, they need a zero which doesn't change what it is added to, and they need to support scalar multiplication (this isn't really required, but it is handy for things like gradient descent). 
+Beyond being a vector space, differentials need to be able to be added to a primal value to get back another primal value. 
 Or roughly equivalently a differential is a difference between two primal values.
 
-One thing to note in this example is that the *primal* does not have to be a vector space.
-We can however always transform it to be a vector space via choosing it an origin and expressing each primal value as a differential typed distance from that origin.
+One thing to note in this example is that the primal does not have to be a vector space.
+We can however always transform it to be a vector space via choosing an origin, and expressing each primal value as a differential typed distance from that origin.
```

Comment:
I can't parse this sentence. 

"We can transform an object t to a vector space" parses to tangent functor maps object ``x`` to tangent spaces ``T_x``.
Now I get stuck. The following discussion does hold for tangent spaces: There is a natural origin in a tangent space, given by the "zero" derivation // curve // dual to the zero function. If we think about exponential maps, ie. maps (I'm lying here: these can exist if you restrict to small neighborhoods) ``T_x \rightarrow X``, the zero always represents the point ``x``.

The choice of ``x`` also reflects the action by the *torsor* group, which happens to be isomorphic to ``X``, in the case of a single Float DateTime type. DateTime is not a vector space, but we have the happy accident that we do have a non-canonical isomorphism between ``T_x X`` and ``X``. 

 It's a bit more tricky if our type is, say, periodic, but we still have a surjective exponential map ``T_x \rightarrow X`` for all ``x``. This map can be written down as a formal power series pretty generally!

Reply:
I am just going to delete that section as it doesn't add much at all, and can confuse things.